a10_octavia/controller/worker/flows/a10_listener_flows.py

```diff
@@ -70,20 +74,74 @@ def get_create_listener_flow(self, topology):
+                requires=a10constants.VTHUNDER,
+                provides=a10constants.VTHUNDER))
+        create_listener_flow.add(self.handle_ssl_cert_flow(flow_type='create', listener=listener))
+        create_listener_flow.add(a10_database_tasks.GetFlavorData(
+            name=sf_name + '_get_flavor',
+            rebind={a10constants.LB_RESOURCE: constants.LISTENER},
+            inject={constants.LISTENER: listener},
+            provides=constants.FLAVOR_DATA))
+        create_listener_flow.add(nat_pool_tasks.NatPoolCreate(
+            name=sf_name + '_natpool_create',
+            requires=(constants.LOADBALANCER,
+                      a10constants.VTHUNDER, constants.FLAVOR_DATA)))
+        create_listener_flow.add(virtual_port_tasks.ListenerCreate(
+            name=sf_name + '_listener_create',
+            requires=[constants.LOADBALANCER, constants.LISTENER,
+                      a10constants.VTHUNDER, constants.FLAVOR_DATA],
+            inject={constants.LISTENER: listener}))
+        create_listener_flow.add(a10_network_tasks.UpdateVIP(
+            name=sf_name + '_update_vip',
+            requires=constants.LOADBALANCER))
+
+        for l7policy in listener.l7policies:
+            create_listener_flow.add(
+                self._l7policy_flows.get_fully_populated_create_l7policy_flow(
+                    topology, listener, l7policy))
+
+        create_listener_flow.add(a10_database_tasks.
+                                 MarkLBAndListenerActiveInDB(
+                                     requires=[constants.LOADBALANCER,
+                                               constants.LISTENER],
+                                     inject={constants.LISTENER: listener}))
+        return create_listener_flow
+
+    def handle_ssl_cert_flow(self, flow_type='create', listener=None):
         if flow_type == 'create':
-            configure_ssl = self.get_ssl_certificate_create_flow()
+            configure_ssl = self.get_ssl_certificate_create_flow(listener)
         elif flow_type == 'update':
-            configure_ssl = self.get_ssl_certificate_update_flow()
+            configure_ssl = self.get_ssl_certificate_update_flow(listener)
         else:
-            configure_ssl = self.get_ssl_certificate_delete_flow(listener_name)
+            configure_ssl = self.get_ssl_certificate_delete_flow(listener)
 
         configure_ssl_flow = graph_flow.Flow(
             a10constants.LISTENER_TYPE_DECIDER_FLOW)
-        check_ssl = cert_tasks.CheckListenerType(
-            name='check_listener_type_' + listener_name,
-            requires=constants.LISTENER,
-            rebind={constants.LISTENER: listener_name})
+
+        if listener is not None:
+            check_ssl = cert_tasks.CheckListenerType(
+                name='check_listener_type_' + listener.id,
+                requires=constants.LISTENER,
+                inject={constants.LISTENER: listener})
+        else:
```

Comment:
Do we need this if-else check 

As per my understanding we will come to this flow if listener is present in listeners.

https://github.com/a10networks/a10-octavia/pull/574/files#diff-069d94c635a94902be690b4ff65292d39b780e0ccc6e2ce86afb425c882a7226R131

Reply:
yes, handle_ssl_cert_flow() still called by other follows (other than fully populated flow).
For example: get_create_listener_flow() it will not provide the listener when calling handle_ssl_cert_flow() and the behavior should be the same as before.